# R Analysis Practice

### Clinical and laboratory measurements are performed on patients hospitalized in normal wards and intensive care wards in a hospital. It is being investigated whether these measurements differ in patients hospitalized in different wards. It is requested to analyze the kurtosis and skewness coefficient of the data.

In [2]:
library(readxl)
library(tidyverse)
library(EnvStats)

In [3]:
df <- read_excel("/data/notebook_files/Hospital_Data.xlsx")

In [4]:
head(df)

A tibble: 6 × 2 
 
 Normal_Wards Intensive_Care_Wards 
 <dbl> <dbl> 
 
 
 152 217 
 134 180 
 143 153 
 135 114 
 208 160 
 114 186

In [5]:
glimpse(df)

Rows: 40
Columns: 2
$ Normal_Wards         <dbl> 152, 134, 143, 135, 208, 114, 165, 226, 107, 223,…
$ Intensive_Care_Wards <dbl> 217, 180, 153, 114, 160, 186, 128, 180, 123, 109,…


In [41]:
skewness_normal <- skewness(df$Normal_Wards)
skewness_normal

1.23460181612816

In [42]:
kurtosis_normal <- kurtosis(df$Normal_Wards)
kurtosis_normal

1.32238215782059

The skewness coefficient for normal wards is 1.234 and the kurtosis coefficient is 1.322. Accordingly, it can be said that the skewness of the normal wards is in the positive direction (right-skewed) and the kurtosis is steeper than the normal distribution. It can be inferred that the distribution is significantly skewed

In [43]:
skewness_intensive <- skewness(df$Intensive_Care_Wards)
skewness_intensive

1.03481411563784

In [44]:
kurtosis_intensive <- kurtosis(df$Intensive_Care_Wards)
kurtosis_intensive

0.678039863989537

The skewness coefficient for the intensive care wards is 1.034 and the kurtosis coefficient is 0.678. Accordingly, it can be said that the skewness of the intensive care wards is in the positive direction (skewed to the right) and the kurtosis is flatter than the normal distribution.

#### Bowley and Pearson Asymmetry Measures;

##### Normal Wards

In [10]:
quantile(df$Normal_Wards)

0% 100 25% 122.75 50% 144 75% 187.25 100% 293

In [21]:
Q1_nor<- 122.75
Q2_nor<- 144
Q3_nor<- 187.25
bow_nor <- (Q3_nor+Q1_nor-2*Q2_nor)/(Q3_nor-Q1_nor)
cat("Bowley Asymmetry Measure of Normal Wards:",bow_nor)

Bowley Asymmetry Measure of Normal Wards: 0.3410853

The Bowley Asymmetry Measure of normal wards is 0.341. Accordingly, since the value is positive, the distribution is skewed to the right.

In [35]:
mean_normal <- mean(df$Normal_Wards)
median <- median(df$Normal_Wards)
sd_normal <- sd(df$Normal_Wards)
mode_df <- table(df$Normal_Wards)
mode_normal <- as.numeric(names(mode_df)[which.max(mode_df)])
pear_nor <- (mean_normal-mode_normal)/sd_normal
cat("Pearson Asymmetry Measure of Normal Wards:",pear_nor)

Pearson Asymmetry Measure of Normal Wards: 1.058656

Pearson Asymmetry Measure of normal wards is 1.058. Accordingly, as in the Bowley Measure, the distribution is skewed to the right as the value is positive. 

##### Intensive_Care_Wards

In [36]:
quantile(df$Intensive_Care_Wards)

0% 101 25% 118.75 50% 151 75% 186.5 100% 291

In [37]:
Q1_nor<- 118.75
Q2_nor<- 151
Q3_nor<- 186.5
bow_nor <- (Q3_nor+Q1_nor-2*Q2_nor)/(Q3_nor-Q1_nor)
cat("Bowley Asymmetry Measure of Intensive Care Wards:",bow_nor)

Bowley Asymmetry Measure of Intensive Care Wards: 0.04797048

The Bowley Asymmetry Measure of intensive care wards is 0.0479. Accordingly, since the value is positive, the distribution is skewed to the right

In [38]:
mean_normal <- mean(df$Intensive_Care_Wards)
median <- median(df$Intensive_Care_Wards)
sd_normal <- sd(df$Intensive_Care_Wards)
mode_df <- table(df$Intensive_Care_Wards)
mode_normal <- as.numeric(names(mode_df)[which.max(mode_df)])
pear_nor <- (mean_normal-mode_normal)/sd_normal
cat("Pearson Asymmetry Measure of Intensive Care Wards:",pear_nor)

Pearson Asymmetry Measure of Intensive Care Wards: 0.9098589

Pearson Asymmetry Measure of normal wards is 0.909. Accordingly, as in the Bowley Measure, the distribution is skewed to the right as the value is positive.